### PR TITLE
Feature: text-to-answer api

### PR DESCRIPTION
**Pull request by wwwy3y3**
###  Mutation
```graphql
# Generate Thread Response Breakdown
  generateThreadResponseBreakdown(
    threadId: Int!
    responseId: Int!
  ): ThreadResponse!

  # Generate Thread Response Answer
  generateThreadResponseAnswer(
    threadId: Int!
    responseId: Int!
  ): ThreadResponse!
```

### Query
```graphql
type ThreadResponseAnswerDetail {
    queryId: String
    status: String
    error: Error
    numRowsUsedInLLM: Int
    content: String
  }

  type ThreadResponseBreakdownDetail {
    queryId: String
    status: AskingTaskStatus!
    error: Error
    description: String
    steps: [DetailStep!]!
  }

  type ThreadResponse {
    id: Int!
    threadId: Int!
    question: String!
    sql: String!
    view: ViewInfo
    breakdownDetail: ThreadResponseBreakdownDetail
    answerDetail: ThreadResponseAnswerDetail
  }
```

## Streaming API
```
/api/ask_task/streaming_answer
```

### CODE REVIEW

File: wren-ui/src/pages/api/ask_task/streaming_answer.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+  // Method to remove content from the map
+  public remove(key: string) {
+    delete this.contentMap[key];
+  }
+}
+
+const contentMap = new ContentMap();
+
+export default async function handler(
```
Comment (onlyjackfrost) [inline wren-ui/src/pages/api/ask_task/streaming_answer.ts:31]: should we restrict the request method?

**Review by onlyjackfrost — COMMENTED** [review]
besides some questions, other LGTM

File: wren-ui/src/apollo/server/services/askingService.ts [hunk 2]
```diff
@@ -304,14 +335,153 @@ class BackgroundTracker {
   }
 }
 
+class TextBasedAnswerBackgroundTracker {
```
Comment (onlyjackfrost) [inline wren-ui/src/apollo/server/services/askingService.ts:338]: Could we move the BackgroundTracker related class under the `server/backgrounds` folder?

File: wren-ui/src/apollo/server/services/askingService.ts [hunk 3]
```diff
@@ -581,31 +745,103 @@ export class AskingService implements IAskingService {
...
+  ): Promise<ThreadResponse> {
+    const { language } = configurations;
+    const thread = await this.threadRepository.findOneBy({
+      id: threadId,
+    });
+    const threadResponse = await this.threadResponseRepository.findOneBy({
+      id: threadResponseId,
+    });
+
+    if (!thread) {
```
Comment (onlyjackfrost) [inline wren-ui/src/apollo/server/services/askingService.ts:779]: I think checking the threadResponse is enough here. 
I found @andreashimin use the same pattern in the text-to-chart PR and I ask [the same question](https://github.com/Canner/WrenAI/pull/997#discussion_r1886921822) there.

File: wren-ui/src/pages/api/ask_task/streaming_answer.ts [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    });
+
+    stream.on('end', () => {
+      res.write(`data: ${JSON.stringify({ done: true })}\n\n`);
+      res.end();
+      askingService
+        .changeThreadResponseAnswerDetailStatus(
+          Number(responseId),
+          ThreadResponseAnswerStatus.FINISHED,
+          contentMap.getContent(queryId),
```
Comment (onlyjackfrost) [inline wren-ui/src/pages/api/ask_task/streaming_answer.ts:88]: Could you kindly add the posthog event for telemetry here?

File: wren-ui/src/apollo/server/resolvers/askingResolver.ts [hunk 5]
```diff
@@ -352,14 +353,10 @@ export class AskingResolver {
...
 
-    const project = await ctx.projectService.getCurrentProject();
     const askingService = ctx.askingService;
     const eventName = TelemetryEvent.HOME_ASK_FOLLOWUP_QUESTION;
     try {
-      const response = await askingService.createThreadResponse(data, {
-        threadId,
-        language: WrenAILanguage[project.language] || WrenAILanguage.EN,
-      });
+      const response = await askingService.createThreadResponse(data, threadId);
```
[the remark this answers is not in the document]
Comment (wwwy3y3) [reply wren-ui/src/apollo/server/resolvers/askingResolver.ts:359]: In this branch, creating a thread or a thread response will not call `askingService.generateAskDetail` anymore. Generating ask detail will be done in `generateThreadResponseBreakdown` mutation. Thus, we don't need to pass in language of project anymore.

File: wren-ui/src/pages/api/ask_task/streaming_answer.ts (hunk 1 above)
Comment (wwwy3y3) [reply wren-ui/src/pages/api/ask_task/streaming_answer.ts:32]: resolved in [2b34fee](https://github.com/Canner/WrenAI/pull/986/commits/2b34fee36fc94df587dceb8e93ee6ce53b846efa)

File: wren-ui/src/apollo/server/services/askingService.ts (hunk 2 above)
Comment (wwwy3y3) [reply wren-ui/src/apollo/server/services/askingService.ts]: resolved in [fcd5236](https://github.com/Canner/WrenAI/pull/986/commits/fcd5236dfa598c2982df1e18230c2f91fc61c8ad)

File: wren-ui/src/apollo/server/services/askingService.ts (hunk 3 above)
Comment (wwwy3y3) [reply wren-ui/src/apollo/server/services/askingService.ts]: resolved in [fa383c9](https://github.com/Canner/WrenAI/pull/986/commits/fa383c92c402135cfcd9196a0bf638966dd7ca2d)

File: wren-ui/src/pages/api/ask_task/streaming_answer.ts (hunk 4 above)
Comment (wwwy3y3) [reply wren-ui/src/pages/api/ask_task/streaming_answer.ts:94]: resolved in [b4a0cac](https://github.com/Canner/WrenAI/pull/986/commits/b4a0cac05524cda78804d4dc78401d98e6416d9f)

**Review by onlyjackfrost — APPROVED** [review]

**Pull request merged**